pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1421,6 +1421,94 @@ func TestServiceRegistryExternalService(t *testing.T) {
 	}
 }
 
+func TestAllocateLoadBalancerNodePortsFalse(t *testing.T) {
+	ctx := genericapirequest.NewDefaultContext()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceLBNodePortControl, true)()
+
+	storage, registry, server := NewTestREST(t, nil, singleStackIPv4)
+	defer server.Terminate(t)
+
+	svc := &api.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "alloc-false"},
+		Spec: api.ServiceSpec{
+			AllocateLoadBalancerNodePorts: &[]bool{false}[0],
+			Selector:                      map[string]string{"bar": "baz"},
+			SessionAffinity:               api.ServiceAffinityNone,
+			Type:                          api.ServiceTypeLoadBalancer,
+			Ports: []api.ServicePort{{
+				Port:       6502,
+				Protocol:   api.ProtocolTCP,
+				TargetPort: intstr.FromInt(6502),
+			}},
+		},
+	}
+	_, err := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("Failed to create service: %#v", err)
+	}
+	srv, err := registry.GetService(ctx, svc.Name, &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if srv == nil {
+		t.Fatalf("Failed to find service: %s", svc.Name)
+	}
+	serviceNodePorts := collectServiceNodePorts(srv)
+	if len(serviceNodePorts) != 0 {
+		t.Errorf("Allocated NodePorts with AllocateLoadBalancerNodePorts=false : %s", srv.Name)
+	}
+
+	for i := range serviceNodePorts {
+		nodePort := serviceNodePorts[i]
+		// Release the node port at the end of the test case.
+		storage.serviceNodePorts.Release(nodePort)
+	}
+}
+
+func TestAllocateLoadBalancerNodePortsTrue(t *testing.T) {
+	ctx := genericapirequest.NewDefaultContext()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceLBNodePortControl, true)()
```

Comment:
I think we also want a test case where the gate is false and so setting `AllocateLoadBalancerNodePorts` is a no-op.